apps/myaccount/src/components/consents/consent-edit.tsx

```diff
@@ -50,38 +58,194 @@ export const AppConsentEdit: FunctionComponent<EditConsentProps> = (
+            editingConsent.consentReceipt &&
+            editingConsent.consentReceipt.services &&
+            editingConsent.consentReceipt.services.length > 0;
+    }
+
+    /**
+     * A predicate that checks whether a particular service inside
+     * a {@link ConsentReceiptInterface} has purposes to show.
+     *
+     * @param {ServiceInterface} service
+     */
+    const hasPurposesInService = (service: ServiceInterface): boolean => {
+        return service &&
+            service.purposes &&
+            service.purposes.length > 0;
+    }
+
+    /**
+     * A predicate that checks whether a particular {@link PurposeInterface}
+     * has a list of {@link PIICategory[]} and is not empty.
+     *
+     * @param {PurposeInterface} purpose
+     */
+    const hasPIICategoriesInPurpose = (purpose: PurposeInterface): boolean => {
+        return purpose.piiCategory && purpose.piiCategory.length > 0;
+    }
+
+    /**
+     * JSX Builder method. Invoked within the scope of {@link this}
+     *
+     * @param {PurposeInterface} purpose
+     */
+    const eachPurposeToJSX = (purpose: PurposeInterface) => {
+
+        /**
+         * JSX Builder method. Invoked within the scope of {@link eachPurposeToJSX}
+         * @param {PIICategoryWithStatus} piiCat
+         */
+        const eachPIICategoryItem = (piiCat: PIICategoryWithStatus) => {
+            return <List.Item key={ piiCat.piiCategoryId }>
+                <List.Content>
+                    <List.Header>
+                        <Checkbox
+                            className={ isRevoked(
+                                piiCat.piiCategoryId,
+                                purpose.purposeId,
+                                editingConsent.consentReceiptID
+                            ) ? "revoked" : "" }
+                            checked={ !isRevoked(
+                                piiCat.piiCategoryId,
+                                purpose.purposeId,
+                                editingConsent.consentReceiptID
+                            ) }
+                            label={ piiCat.piiCategoryDisplayName }
+                            onChange={ () => onPIIClaimToggle(
+                                piiCat.piiCategoryId,
+                                purpose.purposeId,
+                                editingConsent.consentReceiptID
+                            ) }
+                        />
+                        {
+                            isRevoked(piiCat.piiCategoryId, purpose.purposeId, editingConsent.consentReceiptID) ? (
+                                <Label className="revoked-label" horizontal>
+                                    { t("common:revoked") }
+                                </Label>
+                            ) : null
+                        }
+
+                    </List.Header>
+                </List.Content>
+            </List.Item>
+        };
+
+        return ( <React.Fragment key={ purpose.purposeId }>
```

Comment:
Check the ending as well.
```suggestion
        return (
            <React.Fragment key={ purpose.purposeId }>
```

Reply:
This has been resolved in commit edd7c54